persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/impl/table/AbstractTableManager.java

```diff
@@ -685,11 +685,14 @@ public void upsertEntry(Connection connection, int segment, MarshallableEntry<?
 
    @Override
    protected void prepareValueStatement(PreparedStatement ps, int segment, MarshallableEntry<? extends K, ? extends V> entry) throws SQLException {
-      String keyStr = key2Str(entry.getKey());
-      ByteBuffer byteBuffer = marshall(entry.getMarshalledValue(), marshaller);
-      ps.setBinaryStream(1, new ByteArrayInputStream(byteBuffer.getBuf(), byteBuffer.getOffset(),
-            byteBuffer.getLength()), byteBuffer.getLength());
-      ps.setLong(2, entry.expiryTime());
+      prepareValueStatement(ps, segment, entry.getKey(), marshall(entry.getMarshalledValue(), marshaller), entry.expiryTime());
+   }
+
+   protected void prepareValueStatement(PreparedStatement ps, int segment, K key, ByteBuffer valueBytes, long expiryTime) throws SQLException {
```

Comment:
I'm not spotting why you need a new method. 
The code seems the same...

Reply:
It is not, notice how it is calling marshall and passing the ByteBuffer itself. The DB2 and Oracle TableManagers were not marshalling the value thus inserting a corrupted value.

Reply:
I added this method so extensions don't have to marshall. I would have liked to make the other method final, but cannot as the SQL store does its own marshalling.

Reply:
shouldn't the `key` be a `String`?

Reply:
No, not yet. It is converted to a String one line below, however I could do this in the other method. The SQL Stores also do their own conversions so it can't be defined as String for the other method either.